src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -3208,7 +3209,7 @@ public override bool Execute()
             return Execute(
                 p => FileUtilities.FileExistsNoThrow(p),
                 p => FileUtilities.DirectoryExistsNoThrow(p),
-                (p, searchPattern) => Directory.GetDirectories(p, searchPattern),
+                (p, searchPattern) => FileSystems.Default.EnumerateDirectories(p, searchPattern).ToArray(),
```

Comment:
If the collection created here is short lived, `ToList()` may allocate less than `ToArray()`, as it doesn't have to trim the buffer to the right size.

Reply:
Wasn't the main perf gain in the fact that we avoided enmurating all the matching directories on the disk (and with `Any()` we could break early)?

Reply:
This is good advice but since it'd require changing the signature I'm willing to wait to do it.